src/include/http_server.hpp

```diff
@@ -1,596 +1,54 @@
-     * at different points in the life cycle of a req. (like on_headers etc) These hooks from evhtp library do not seem
-     * to work properly when the hook cb functions return anything other than EVHTP_RES_OK For a perfect implementation
-     * that avoids users to add http_auth_verify before all the apis they define, we need to either explore evhtp lib
-     * more or switch to a different server like Pistache.
-     */
-
-    evhtp_res http_auth_verify(evhtp_request_t* req, std::string& msg) {
-        if (!m_cfg.is_auth_enabled) { return EVHTP_RES_OK; }
-
-        const std::string bearer{"Bearer "};
-        auto* token{::evhtp_header_find(req->headers_in, "Authorization")};
-        if (!token) {
-            msg = "missing auth token in request header";
-            LOGDEBUGMOD(httpserver_lmod, "Processing req={}; {}", static_cast< void* >(req), msg);
-            return EVHTP_RES_UNAUTH;
-        }
-        const std::string token_str{token};
-        if (token_str.rfind(bearer, 0) != 0) {
-            msg = "require bearer token in request header";
-            LOGDEBUGMOD(httpserver_lmod, "Processing req={}; {}", static_cast< void* >(req), msg);
-            return EVHTP_RES_UNAUTH;
-        }
-        const auto raw_token{token_str.substr(bearer.length())};
-        // verify method is expected to not throw
-        return to_evhtp_res(m_auth_mgr->verify(raw_token, msg));
-    }
-
-#define request_callback(cb)                                                                                           \
-    (evhtp_callback_cb) std::bind(&HttpServer::cb, this, std::placeholders::_1, std::placeholders::_2)
-#define error_callback(cb)                                                                                             \
-    (evhtp_hook) std::bind(&HttpServer::cb, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3)
-
-protected:
-    /* ****************** All default connection handlers ****************** */
-    static evhtp_res register_connection_handlers(evhtp_connection_t* conn, void* arg) {
-        evhtp_connection_set_hook(conn, evhtp_hook_on_path, (evhtp_hook)HttpServer::request_on_path_handler, arg);
-        evhtp_connection_set_hook(conn, evhtp_hook_on_request_fini, (evhtp_hook)HttpServer::request_fini_handler, arg);
-        evhtp_connection_set_hook(conn, evhtp_hook_on_conn_error, (evhtp_hook)HttpServer::connection_error_callback,
-                                  arg);
-        evhtp_connection_set_hook(conn, evhtp_hook_on_error, (evhtp_hook)HttpServer::request_error_handler, arg);
-        return EVHTP_RES_OK;
-    }
-
-    static void default_request_handler(evhtp_request_t* req, void* arg) {
-        HttpServer* const server{static_cast< HttpServer* >(arg)};
-        const HttpCallData cd{new _http_calldata(req, arg)};
-        server->respond_NOTOK(cd, EVHTP_RES_BADREQ, "Request can't be matched with any handlers\n");
-    }
+    enum class url_type {
```

Comment:
If you have include sisl/enum here, might as well use sisl ENUM which gives more meaningful logs instead of numbers.

```suggestion
ENUM(url_type, uint8_t,
    localhost,  // url can only be called from localhost
    safe,       // Can be called from any host
    regular
);
```